### PR TITLE
Accept optional predicted usage authorize and oauth_authorize calls

### LINKED ISSUE #28: Can't pass a predicted usage to the authorize call
The 3scale service management API accepts a predicted usage field in the authorize calls, but this gem does not.

### PR DESCRIPTION
**Pull request by davidor**
Closes #28

### CODE REVIEW

**Comment by unleashed** [comment]
Functionality wise this should be ok. But can't stop thinking we could use the opportunity for some refactoring.

File: lib/3scale/client.rb [hunk 1]
```diff
@@ -285,6 +287,13 @@ def encode_transactions(transactions)
       result
     end
 
+    def usage_query_params(usage)
+      usage.map do |metric, value|
+        escaped_metric = CGI.escape "[usage][#{metric}]"
+        "#{escaped_metric}=#{CGI.escape(value.to_s)}"
+      end.join('&')
```
Comment (mikz) [inline lib/3scale/client.rb]: Imo `URI.encode_www_form` could be help here. 

```ruby
URI.encode_www_form(usage.map { |metric, value| ["[usage][#{metric}]", value ] })
```

**Comment by mikz** [comment]
Yeah. Maybe a tiny bit of refactoring would not hurt. But just on the touched lines :)

 Otherwise 👍

File: lib/3scale/client.rb (hunk 1 above)
Comment (davidor) [inline lib/3scale/client.rb]: I just extracted the method as it was. But yes, while we are at it I can change it. Your solution looks cleaner.

**Comment by mayorova** [comment]
It was surprising for me that `app_id=foo&[usage][metric1]=1&[usage][metric2]=2` apparently works the same way that `app_id=foo&usage[metric1]=1&usage[metric2]=2`, but OK :)

**Comment by davidor** [comment]
@mikz I addressed your comment and rebased.

@mayorova Yes, the original code was using the `[usage]` version, I didn't change that.

@unleashed I thought the same. There are some opportunities for refactoring, but I think it's better to tackle that in a separate PR when we have the time, as @mikz suggested, because any refactoring that we could make would probably affect many other things, not just the ones changed in this PR.

**Pull request merged**